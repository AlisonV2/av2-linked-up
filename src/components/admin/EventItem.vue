<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ event.name }}
    </div>
    <p class="card-text">{{ event.description }}</p>
    <p class="card-text">{{ event.localisation }}</p>
    <div class="card-footer text-muted">
      {{ event.date }}
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} event - event object
 * @vue-data {string} role - User role
 */
export default {
  name: 'eventItem',
  props: ['event'],
  data() {
    return {
      role: '',
    };
  },
  created() {
    this.$store
      .dispatch('getUserRole')
      .then(() => {
        this.role = this.$store.getters.getUserRole;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
