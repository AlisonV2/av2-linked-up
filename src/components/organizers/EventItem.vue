<template>
  <div class="card Event-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title" :data-test="event.id">{{ event.name }}</h5>
      <p class="card-text">{{ event.city }} ({{ event.zip }})</p>
    </div>
    <img
      :src="event.thumbnail"
      class="card-img-top"
      :alt="event.id"
    />
    <div class="card-body">
      <h5 class="card-title">{{ event.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} events
 */
export default {
  name: 'EventItem',
  props: ['event'],
};
</script>

<style lang="scss" scoped>
.event-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.Event-card:hover {
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    display: block;
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
