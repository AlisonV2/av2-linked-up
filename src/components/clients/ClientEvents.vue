<template>
  <app-title>Events you plan to attend</app-title>
  <div class="row client-events">
    <div
      class="col-12 col-md-6 col-lg-4 col-xl-3"
      v-for="event in events"
      :key="event.id"
    >
      <router-link :to="{ name: 'Event', params: { id: event.id } }">
        <EventItem :event="event" />
      </router-link>
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';
import EventItem from '@/components/organizers/EventItem';

/**
 * @exports ClientEvents
 * @type {Component}
 */
export default {
  name: 'ClientEvents',
  components: {
    EventItem,
  },
  data() {
    return {
      events: [],
    };
  },
  created() {
    this.$store
      .dispatch('getClientEvents')
      .then(() => {
        this.events = this.$store.getters.getClientEvents;
        console.log(this.events)
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
