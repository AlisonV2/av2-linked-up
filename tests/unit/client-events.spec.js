import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import ClientEvents from '@/components/clients/ClientEvents.vue';
import EventItem from '@/components/organizers/EventItem';
import AppTitle from '@/components/app/Title';

const actions = {
  getClientEvents: jest.fn(),
};

const getters = {
  getClientEvents: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const $route = {
  name: 'Event',
  params: {
    id: '1'
  }
};

const config = {
  data() {
    return {
      events: [],
    };
  },
  global: {
    plugins: [store],
    components: {
      EventItem,
      AppTitle
    },
    stubs: {
      RouterLink: RouterLinkStub,
    },
    mocks: {
      $route
    }
  },
};

/**
 * @module ClientEventsTest
 */
describe('ClientEvents.vue', () => {
  it('ClientEvents Snapshot', () => {
    const component = shallowMount(ClientEvents, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if getClientEvents action is called', async () => {
    const component = shallowMount(ClientEvents, config);
    ClientEvents.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getClientEvents).toHaveBeenCalled;
  });

  it('Check if getClientEvents getter is called', async () => {
    const component = shallowMount(ClientEvents, config);
    ClientEvents.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getClientEvents).toHaveBeenCalled;
  });
});
