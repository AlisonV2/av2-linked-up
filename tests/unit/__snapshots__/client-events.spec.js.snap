// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientEvents.vue ClientEvents Snapshot 1`] = `
<div
  class="row client-events"
/>
`;
